src/main/java/com/monumental/services/ModelService.java

```diff
@@ -282,33 +301,41 @@ public void delete(List<Integer> ids) throws HibernateException {
     }
 
     /**
-     * Generic method to build a "FROM ... WHERE ..." query
-     * Assumes the FROM clause is the name of the T generic class
-     * @param whereParameterName - Name of the column to use in the WHERE clause
-     * @param whereParameter - Used to set the parameter in the WHERE clause
-     * @return List<T> - the List of T returned by the query
+     * Perform a get query with the specified criteria
+     * @param criteria - List of Criterion objects to apply to the query
+     * @param initializeLazyLoadedCollections - If true, loads all of the collections associated with T that are
+     *                                        normally lazy loaded
+     * @return List<T> - List of T returned by the query
      */
     @SuppressWarnings("unchecked")
-    public List<T> getFromWhere(String whereParameterName, Object whereParameter) {
+    public List<T> getWithCriteria(List<Criterion> criteria, boolean initializeLazyLoadedCollections) {
```

Comment:
You should also add a `getWithCriteria(List<Criterion criteria> criteria)` that calls this method with `initializeLazyLoadedCollections` to false so that it's essentially an optional param. See the other get methods for reference

Reply:
Also this is super nitpicky but I kind of prefer that the method just be called `get` and accept the criteria param. I don't know why but I like them all having the same name